### PR TITLE
Add function to query when next operation is due

### PR DESCRIPTION
**Pull request by jcheng5**

### CODE REVIEW

**Comment by jcheng5** [comment]
@wch This is needed for shiny async. Do you mind reviewing when you get a chance?

File: R/later.R [hunk 1]
```diff
@@ -104,3 +104,14 @@ run_now <- function() {
...
 }
+
+#' Relative time to next scheduled operation
+#'
+#' Returns the duration between now and the earliest operation that is currently
+#' scheduled, in seconds. If the operation is in the past, the value will be
+#' negative. If no operation is currently scheduled, the value will be `Inf`.
+#'
+#' @export
+next_op_secs <- function() {
```
Comment (wch) [inline R/later.R]: Is it necessary to have a wrapper function, instead of just naming the C++ function `next_op_secs`?

File: src/later.cpp [hunk 2]
```diff
@@ -81,6 +81,17 @@ void execLater(Rcpp::Function callback, double delaySecs) {
...
 }
 
+// [[Rcpp::export]]
+double secsToNext() {
+  Optional<Timestamp> nextTime = callbackRegistry.nextTimestamp();
+  if (!nextTime.has_value()) {
+    return R_PosInf;
+  } else {
+    Timestamp now;
+    return (*nextTime).diff_secs(now);
```
Comment (wch) [inline src/later.cpp]: Is this syntax needed instead of `->` because of the way the Optional class works?

**Comment by wch** [comment]
Other than a couple questions that I have, it looks good.

File: src/later.cpp (hunk 2 above)
Comment (jcheng5) [reply src/later.cpp]: Yeah. I implemented the `*` operator but not the `->` operator--I probably could/should.

File: R/later.R (hunk 1 above)
Comment (jcheng5) [reply R/later.R]: Oh, I guess that works? And just write the roxygen in the .cpp file?

Comment (wch) [reply R/later.R]: Yup, that's what i had in mind.

**Pull request merged**